### PR TITLE
[WIP] Community events have display levels

### PR DESCRIPTION
**Pull request by lpil**
trippy-doors

### CODE REVIEW

File: lib/community/communityFields.js [hunk 1]
```diff
@@ -62,7 +56,17 @@ const communityType = new GraphQLObjectType({
       type: GraphQLString,
       description: 'The summary of the community mailing list',
     },
-    events: eventsList('events'), // TODO: Test my resolver!
+    // TODO: Test me!
+    events: {
+      type: new GraphQLList(CommunityEventType),
+      description: 'The community\'s events',
+      resolve(community) {
+        if (community) {
```
Comment (AndrewBestbier) [inline lib/community/communityFields.js]: && community.events

Comment (lpil) [inline lib/community/communityFields.js]: This has been like this for a while so I don't think it's needed.

Comment (lpil) [inline lib/community/communityFields.js]: Can add just to be safe though?

File: lib/community/communityFields.js [hunk 2]
```diff
@@ -11,21 +11,15 @@ import {
 } from '../fetch';
 
 import { EventType } from '../event/eventFields';
+import {
+  DEFAULT_DISPLAY_LEVEL,
```
Comment (AndrewBestbier) [inline lib/community/communityFields.js]: Is this import really necessary? You could just set a default in prismic

File: prismic/custom-types/community.json [hunk 3]
```diff
@@ -56,6 +56,17 @@
...
+          },
+          "displayLevel": {
+            "type" : "Select",
+            "config" : {
+              "label" : "Display Level",
+              "options" : [
+                "Featured",
+                "Highlighted",
+                "Regular"
+              ]
```
Comment (AndrewBestbier) [inline prismic/custom-types/community.json]: Set a default value in prismic?

File: lib/community/communityFields.js (hunk 1 above)
Comment (AndrewBestbier) [inline lib/community/communityFields.js]: Yup maybe just to be safe :)

File: lib/communityEvent/communityEventFields.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import {
+  GraphQLObjectType,
+  GraphQLEnumType,
+} from 'graphql';
+import { eventTypeFields } from '../event/eventFields';
+import EventInterface from '../interfaces/event';
```
Comment (AndrewBestbier) [inline lib/communityEvent/communityEventFields.js]: Much smarter to use interfaces than my original idea.

Comment (lpil) [inline lib/communityEvent/communityEventFields.js]: Thanks @charypar ! :D

File: lib/event/eventFields.js [hunk 5]
```diff
@@ -24,104 +24,110 @@ import {
...
-      resolve(event) {
-        if (event) {
-          return event.talks.map((result) => {
-            const id = pathOr('', ['data', 'value', 'document', 'id'], result);
-            return fetchTalk({ id });
-          });
-        }
-        return [];
-      },
+export const eventTypeFields = {
```
Comment (AndrewBestbier) [inline lib/event/eventFields.js]: This moved as per our offline discussion

File: prismic/custom-types/community.json (hunk 3 above)
Comment (lpil) [inline prismic/custom-types/community.json]: Good idea :)

File: lib/interfaces/event/index.js [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+          const id = pathOr('', ['data', 'value', 'document', 'id'], result);
+          return fetchTalk({ id });
+        });
+      }
+      return [];
+    },
+  },
+};
+
+
```
Comment (bkspace) [inline lib/interfaces/event/index.js]: extra line

File: lib/communityEvent/index.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+export const CommunityEventType = new GraphQLObjectType({
+  name: 'CommunityEvent',
+  interfaces: [
+    EventInterface,
+  ],
+  fields: {
+    ...eventTypeFields,
+    displayLevel: {
+      type: DisplayLevelType,
+      description: 'The extent to which the event is highlighted within the' +
```
Comment (bkspace) [inline lib/communityEvent/index.js]: no space between two strings

File: lib/communityEvent/index.spec.js [hunk 8]
```diff
@@ -0,0 +1 @@
+import '.'; // It doesn't crash.
```
Comment (bkspace) [inline lib/communityEvent/index.spec.js]: Could we not be checking the structure of the communityEvent is setup properly? I.e that displayLevel gets added to normal eventFields?

Comment (lpil) [inline lib/communityEvent/index.spec.js]: There's not really any mechanism for testing that stuff in badger brain yet. I could check the schema has that key but I don't think it's useful. I really wanna test the resolvers and the relationships between them in an integration sense, but it's a big job getting BB ready for that.

**Comment by AndrewBestbier** [comment]
When running tests: BasicEvent may only implement Interface types, it cannot implement: undefined on local?

**Pull request merged**